android/java/org/chromium/chrome/browser/vpn/activities/BraveVpnPlansActivity.java

```diff
@@ -114,6 +114,8 @@ public void onClick(View v) {
         yearlySubscriptionAmountText.setText(
                 String.format(getResources().getString(R.string.yearly_subscription_amount),
                         yearlySkuDetails.getPrice()));
+        BraveVpnUtils.showProgressDialog(
+                BraveVpnPlansActivity.this, getResources().getString(R.string.vpn_connect_text));
         mIsVerification = true;
         verifySubscription();
```

Comment:
Shouldn't we move that to `finishNativeInitialization`? We need native to be initialized as we do JNI calls inside.

Reply:
it's not obvious as it's executed inside `BraveFirstRunFlowSequencer`. Not sure how fast it could be done and is there any extra user's interactions involved. Perhaps would be save to do that in any case after native initialisation is done.